barretenberg/cpp/src/barretenberg/flavor/ultra_recursive.hpp

```diff
@@ -376,6 +404,44 @@ template <typename BuilderType> class UltraRecursive_ {
             this->lagrange_first = verification_key->lagrange_first;
             this->lagrange_last = verification_key->lagrange_last;
         }
+
+        VerifierCommitments(const std::shared_ptr<VerificationKey>& verification_key,
```

Comment:
in the standalone Goblin Honk Verifier we were making an incomplete VerifierCommitments object from vk and then manually assigning the witness commitments, seemed ok to have this option as well since the decider has all data readily available